### PR TITLE
gRPC: don't start HTTP server

### LINKED ISSUE #20334: Make HTTP server optional for gRPC
### Description

When adding `quarkus-grpc` extension as dependency, it pulls `vertx` and `vertx-web` extensions which are overhead for simple applications or projects which don't use WEB at all so Quarkus shouldn't bind and listen to HTTP ports. When someone is using the gRPC extension, it shouldn't assume that WEB is also used. 

### Implementation ideas

The HTTP server should be optional as it is used only for DevUI. This `<!-- for dev mode, TODO: try to get rid of it-->` is present as a comment to the pom.xml of `quarkus-grpc` extension.

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
fixes #20334

### CODE REVIEW

**Comment by michalszynkiewicz** [comment]
the test failures don't look relevant

**Review by kenfinnigan — APPROVED** [review]
Seems ok, but it's been a while since I've looked at this area. Might be worth @ebullient and @stuartwdouglas taking a look to confirm as well.

Is this an approach that Reactive Messaging extensions could take as well? /cc @cescoffier

**Comment by ebullient** [comment]
NonApplicationRoutePathBuildItem is closely related to the others.. should they move as a set, or do we only consider this one SPI? @stuartwdouglas ?

**Comment by michalszynkiewicz** [comment]
just a note, I'm not moving `NonApplicationRoutePathBuildItem` here, I just created a way to configure it bringing in vertx-http, the build item stayed where it was.

File: extensions/vertx-http/deployment-spi/src/main/java/io/quarkus/vertx/http/deployment/spi/NonApplicationRootPathConfigurerBuildItem.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package io.quarkus.vertx.http.deployment.spi;
+
+import java.util.function.Consumer;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+public final class NonApplicationRootPathConfigurerBuildItem extends MultiBuildItem {
```
[the remark this answers is not in the document]
Comment (michalszynkiewicz) [reply extensions/vertx-http/deployment-spi/src/main/java/io/quarkus/vertx/http/deployment/spi/NonApplicationRootPathConfigurerBuildItem.java:7]: as far as I understand they both mean the same, and the former is shorter :)

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]
We already have devconsole SPI artifacts: https://github.com/quarkusio/quarkus/pull/20465

**Pull request closed without merging**